File: solutions/go/vehicle-purchase/vehicle_purchase.go
package purchase

func NeedsLicense(kind string) bool {
	return kind == "car" || kind == "truck"
}

func ChooseVehicle(option1, option2 string) string {
	const better string = " is clearly the better choice."
    if option1 < option2 {
    	return option1 + better
    }
    return option2 + better
}

func CalculateResellPrice(originalPrice, age float64) float64 {
    var discountFactor float64 = 0.5
	if age < 10 {
    	discountFactor = 0.7
    }
    if age < 3 {
    	discountFactor = 0.8
    }
    return discountFactor * originalPrice
}

File: solutions/cpp/vehicle-purchase/1/vehicle_purchase.cpp
#include "vehicle_purchase.h"

namespace vehicle_purchase {

    bool needs_license(std::string kind){
        return kind == "car" || kind == "truck";
    }

    std::string choose_vehicle(std::string option1, std::string option2) {
        std::string chosen_car = option1 < option2 ? option1 : option2;
        return chosen_car += " is clearly the better choice.";
    }

    double calculate_resell_price(double original_price, double age) {
        double resell_price = 0.0;
        if( age < 3 )
            resell_price = 0.8 * original_price;
        else if( age < 10 )
            resell_price = 0.7 * original_price;
        else
            resell_price = 0.5 * original_price;
        return resell_price;
    }

}